pelutils/ds/_linecounter.py

```diff
@@ -0,0 +1,121 @@
+            except StopIteration:
+                continue
+            with open(path) as f:
+                lines[ext][i] += len([line for line in f.readlines() if line.strip()])
+
+    repo.git.checkout(branch)
+    return times, lines
+
+def _fuse_times(all_times: list[np.ndarray]) -> np.ndarray:
+    """ Merges all time arrays together into a single array that is also sorted """
+    n = sum(times.size for times in all_times)
+    times = np.empty(n, dtype=int)
+    idcs = [0] * len(all_times)
+    for i in range(n):
+        current_times = [time[idx] if idx < len(time) else float("inf") for time, idx in zip(all_times, idcs)]
+        min_time = np.argmin(current_times)
+        times[i] = current_times[min_time]
+        idcs[min_time] += 1
+    return times
+
+def _last_initial_zero(values: np.ndarray) -> int:
+    if values[0] != 0:
+        return 0
+    return np.where(values!=0)[0][0] - 1
+
+@click.command()
+@click.argument("repos", nargs=-1)
+@click.option("-e", "--extensions", default=_default_extensions, help="Comma seperated list of file extensions to look for")
+@click.option("-o", "--outplot-path", default="linecount.png")
+def linecounter(repos: list[str], extensions: str, outplot_path: str):
+    extensions = [x.strip() for x in extensions.split(",")]
+    extensions = [x if x.startswith(".") else "." + x for x in extensions]
+    wd = os.getcwd()
+    repo_names, all_times, all_counts = list(), list(), list()
+    for repo in repos:
+        repo_path, __ = get_repo(repo)
+        if repo_path is None:
+            raise ValueError("%s is not a git repository" % repo_path)
+        os.chdir(repo_path)
+
+        try:
+            repo = git.Repo()
+            branch = repo.active_branch
+            times, lines = _count(repo, branch, extensions)
+        except KeyboardInterrupt:
+            # Reset repo
+            repo.git.checkout(branch)
+            raise
+        except:
+            raise
+
+        repo_names.append(os.path.split(repo_path)[-1])
+        all_times.append(times)
+        all_counts.append(lines)
+        os.chdir(wd)
+
+    plt.figure(figsize=figsize_std)
+    for i, (repo_name, times, counts) in enumerate(zip(repo_names, all_times, all_counts)):
+        for ext, l in counts.items():
+            if not l.any():
+                continue
+            non_zero = l != 0
+            non_zero[_last_initial_zero(l)] = 1
+            all_times[i] = times[non_zero]
+            all_counts[i] = l[non_zero]
+            plt.plot(all_times[i], all_counts[i], marker=".", label=ext+(f" ({repo_name})" if len(repo_names) > 1 else ""))
+
+    fused_times = _fuse_times(all_times)
+    xticks = np.linspace(0, len(fused_times)-1, 10, dtype=int)
+    xticklabels = [time.strftime("%d-%m-%Y", time.gmtime(x)) for i, x in enumerate(fused_times) if i in xticks]
+    plt.xticks([fused_times[i] for i in xticks], xticklabels, rotation=60)
```

Comment:
At bruge plt direkte i stedet for figur og akse-kald?
cringgggggg...

Reply:
Du kan selv være tåkrummende